homeassistant/components/yamaha/media_player.py

```diff
@@ -109,7 +110,7 @@ def __init__(self, config: None, discovery_info: None) -> None:
         if discovery_info is not None:
             self.name = discovery_info.get("name")
             self.model = discovery_info.get("model_name")
-            self.ctrl_url = discovery_info.get("control_url")
+            self.ctrl_url = str(discovery_info.get("control_url"))
```

Comment:
If the `"control_url"` key is missing in the dict this will convert the default value None to a string. Is that what we want?

Reply:
Oh right, good question

Reply:
Unless we have deeper knowledge about this integration I'd type `self.ctrl_url` as an optional string and remove the string conversion here.

Reply:
Good point I will submit a PR later today

Reply:
Corrected see #53933

Sorry I should have seen that.

Reply:
Don't be sorry. We're improving things one step at a time. :+1: